apps/web/src/actions/evaluations/create.ts

```diff
@@ -12,28 +15,51 @@ export const createEvaluationAction = authProcedure
     z.object({
       name: z.string(),
       description: z.string(),
-      configuration: z.object({
-        type: z.nativeEnum(EvaluationResultableType),
-        detail: z
-          .object({ range: z.object({ from: z.number(), to: z.number() }) })
-          .optional(),
-      }),
-      metadata: z
-        .object({
+      resultConfiguration: z.discriminatedUnion('type', [
```

Comment:
subtle bug always use discriminatedUnions when typing api endpoints

Reply:
:o What's the difference?

Reply:
union type checks against the first matching element whereas discriminatedUnion type checks against the element that matches the "type" prop. The former can cause unexpected issues such as always matching the first element in the union if it contains all optional elements